src/compiler/types.ts

```diff
@@ -5524,11 +5524,11 @@ export const enum IntersectionFlags {
 // dprint-ignore
 /** @internal */
 export const enum ContextFlags {
-    None           = 0,
-    Signature      = 1 << 0, // Obtaining contextual signature
-    NoConstraints  = 1 << 1, // Don't obtain type variable constraints
-    Completions    = 1 << 2, // Ignore inference to current node and parent nodes out to the containing call for completions
-    SkipBindingPatterns = 1 << 3, // Ignore contextual types applied by binding patterns
+    None                 = 0,
+    Signature            = 1 << 0, // Obtaining contextual signature
+    NoConstraints        = 1 << 1, // Don't obtain type variable constraints
+    IgnoreNodeInferences = 1 << 2, // Ignore inference to current node and parent nodes out to the containing call for, for example, completions

```

Comment:
As shown by this PR, this mechanism is handy for other things than just completions - so I decided to rename it.